Automattic/newspack-plugin
# Update UI components

### All Submissions:


### Changes proposed in this Pull Request:

This PR is pretty big, but it mostly changes CSS variable values and names. Specifically:
* It updates the colour variable values to match the new component Figma file, and updates the names to match.
* It updates the font size variables, and updates their names to match. 
* It updates the variable names and values of the border-radius & modal widths.
* It updates the variable names throughout the SCSS files.
* It adds a couple more examples to the UI Demo page, for warnings, font sizes, and the branded buttons.

Outside of that, it also:
* Updates the styles of the radio button, checkbox, and text inputs to match the components (changes include adjusting the border radius, padding, and focus styles.
* Builds out the WooCommerce overrides styles, to make sure the checkout is picking up the new input and font sizing styles.

In the non-SCSS, non-demo files, the PR also updates a couple CSS classes in the reader-activation success message and the reader activation login form.

Still to do:
* Set up the different button size styles.
* Updating the ‘checkbox + label + container’ element.
* Update the label element to incorporate the error styles.

See: 1205234045751551-as-1206899964006422

### How to test the changes in this Pull Request:

1. Start on a test site running RAS.
2. Add a donate block to a page.
3. Create a variable product, and add a checkout button block for it to the same page. 
4. Apply the PR and run `npm run build`. 
5. While logged into a test site, append `?ui-demo` to any page’s URL; review that against the Figma file for font sizes, padding, spacing. 
6. Focus on inputs, and click radio buttons. 
7. Click the ‘Open Modal’ button at the bottom and review its appearance. 
8. View the site in “incognito mode”/as a reader. 
9. Run through the Donate process, and spot check the appearance of the modals.
10. Run through the checkout process using the Checkout button, adn spot check the appearance of the variable modal. 
11. Click the Sign In button in the header, and spot-check the appearance of the modal.
12. Test the 'prefers contrast' colours. In Chrome, you can do this by opening the Developer tools, enabling the Rendering tab, and finding the "Emulate CSS media feature prefers-contrast". Change this to prefers-contrast: more:

![image](https://github.com/Automattic/newspack-plugin/assets/177561/ccb1c1b0-dee8-4023-8a86-90dfa91a2fa2)



### Other information:
